examples/manifold/plot_tsne_faces.py

```diff
@@ -0,0 +1,148 @@
+"""
+===================================
+Comparison of t-SNE methods and options
+===================================
+
+A comparison of t-SNE applied to the Olivetti faces data. The 'exact' and
+'barnes-hut' methods are visualized, as well as init='pca' and init='random'.
+In addition to varying method & init, the script also compares preprocessing
+the input data by reducing the dimensionality via PCA and also varying the
+perplexity.
+
```

Comment:
I think this example is too crowded / long and we don't get the intended message of the example. I would rather simplify it by just demonstrating the cases with the Barnes Hut method and the 3 following configurations:

- random init on raw features
- PCA init on raw features
- PCA init on PCA reduced features (via a pipeline).

The in the text description of the example I would tell that:

- all the three configurations yield similar arrangements of the data,
- PCA-reduced features is interesting to speed up the analysis while preserving most of the pairwise distances of the original feature space. In particular the BallTree data structure used to compute the neigborhoods in the original feature space is known to work faster than brute force only for lower dimensional space (e.g. less than a couple hundreds of dimensions).
- PCA init is interesting to get global arrangement of the groups that are slightly more consistent with the long range dependencies between groups (although I am not 100% sure about this so maybe we should not mention that all).